.github/workflows/test_e2e_deploy_related.yml

```diff
@@ -0,0 +1,71 @@
+name: Test E2E (Vercel Deploy), related
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+env:
+  CARGO_PROFILE_RELEASE_LTO: 'true'
+  DATADOG_API_KEY: ${{ secrets.DATA_DOG_API_KEY }}
+  DD_ENV: 'ci'
+  NAPI_CLI_VERSION: 2.16.2
+  NEXT_JUNIT_TEST_REPORT: 'true'
+  NEXT_TELEMETRY_DISABLED: 1
+  NEXT_TEST_CONTINUE_ON_ERROR: 1
+  NEXT_TEST_JOB: 1
+  NEXT_TEST_MODE: 'deploy'
+  NODE_LTS_VERSION: 20
+  TEST_TIMINGS_TOKEN: ${{ secrets.TEST_TIMINGS_TOKEN }}
+  TURBO_REMOTE_ONLY: 'true'
+  TURBO_TEAM: 'vercel'
+  TURBO_VERSION: 1.12.5
+  VERCEL_TEST_TEAM: vtest314-next-e2e-tests
+  VERCEL_TEST_TOKEN: ${{ secrets.VERCEL_TEST_TOKEN }}
+
+jobs:
+  test:
+    if: github.repository_owner == 'vercel'
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+
+    steps:
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ env.NODE_LTS_VERSION }}
+          check-latest: true
+
+      - name: Enable corepack
+        run: corepack enable
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 25
+
+      - name: Setup tests
+        run: |
+          git fetch origin canary:canary
+          pnpm install
+          pnpm run build
+          npm i -g vercel@latest
+          node scripts/run-e2e-test-project-reset.mjs
+
+      - name: Run tests
+        run: xvfb-run node run-tests.js --related
```

Comment:
If we're using the GitHub runner then we still need to ensure playwright system deps are installed properly or we use their docker image